netbox/dcim/views.py

```diff
@@ -590,9 +590,16 @@ def get(self, request):
         total_count = racks.count()
 
         # Determine ordering
-        reverse = bool(request.GET.get('reverse', False))
-        if reverse:
-            racks = racks.reverse()
+        sort = request.GET.get('sort', "name")
+        if sort:
+            if sort == 'name':
+                racks = racks.order_by('name')
+            elif sort == '-name':
+                racks = racks.order_by('-name')
+            elif sort == 'facility':
+                racks = racks.order_by('facility_id')
+            elif sort == '-facility':
+                racks = racks.order_by('-facility_id')
```

Comment:
We can replace the switch-like block here with a dictionary mapping. This can be used both for validation and for displaying a human-friendly label for the selection.

```suggestion
        ORDERING_CHOICES = {
            'name': 'Name (A-Z)',
            '-name': 'Name (Z-A)',
            'facility_id': 'Facility ID (A-Z)',
            '-facility_id': 'Facility ID (Z-A)',
        }

        sort = request.GET.get('sort')
        if sort not in ORDERING_CHOICES:
            sort = 'name'
```